Source/WebKit/UIProcess/XR/ios/PlatformXRARKit.mm

```diff
@@ -238,6 +244,16 @@
             break;
 
         auto& active = *maybeActive;
+        if (WTF::atomicLoad(&active.terminateRequested))
+            break;
+
+        if ([active.presentationSession sessionEndRequested]) {
+            callOnMainRunLoop([this]() {
```

Comment:
these sort of cross-thread `this` captures cannot be made as I don't think ARKitCoordinator is thread-safe refcounted. So it might already be destroyed at the time the callback gets run.

Reply:
The ARKitCoordinator is a `static LazyNeverDestroyed<ARKitCoordinator> xrCoordinator;` so I hope that it's never destroyed at the time the callback gets run.